### PR TITLE
Update settings manangement, object orient endpoint and function import structure

### PR DESCRIPTION
**Pull request by SerRichard**
…int basic test - accept reformatting

should resolve the issues around the conformance endpoint you have noticed

### CODE REVIEW

**Comment by SerRichard** [comment]
Also, I have merged the test api files together in this :)

**Comment by SerRichard** [comment]
So, added  an abc called EndpointRegister, the idea here is that each Register class will hold the relevant functions, and the Endpoint models attached, this can later be used to derive the "active" endpoints of the api, without the need for the user to define them when init'ing the OpenEOCore. The idea of the Register is to offer a similar modularization of the FastApi routers, but without having to actually use them in any way.

Additionally, I had to update the settings management because of the reason mentioned in teams, I.E, init'ing them in the module code means those are set when the module is imported, this is okay for an application, but not really for a package.

Also updated unit tests to make sure the endpoints called are actually behind the OPENEO_PREFIX value, as they all should be, besides get capabilities :) 

Happy to call and discuss this structure if there are any questions, the only other thing I might add is a RootRegister, for those endpoints that sit on the root path of the api (if you find the Register structure aggreable).

File: openeo_fastapi/client/collections.py [hunk 1]
```diff
@@ -1,69 +1,87 @@
...
-    """
-    stac_url = (
-        app_settings.STAC_API_URL
-        if app_settings.STAC_API_URL.endswith("/")
-        else app_settings.STAC_API_URL + "/"
-    )
+    def _initialize_endpoints(self) -> list[Endpoint]:
+        return [
+            Endpoint(
+                path="/collections",
```
Comment (geospatial-roman) [inline openeo_fastapi/client/collections.py:20]: do we need `/{self.client.settings.OPENEO_VERSION}/` in the path here as well?

**Review by geospatial-roman — APPROVED** [review]
Initialising the settings when the Client is initialised makes sense. In general this looks much tidier than before, thanks! Looks good to me

File: openeo_fastapi/client/collections.py (hunk 1 above)
Comment (SerRichard) [reply openeo_fastapi/client/collections.py:20]: Good shout, but I don't think here we want to add it. This is just about setting what we advertise in /capabilities, and we don't actually want the version number to be present when displayed as an output there. Additionally, I initially thought the same when doing the openeo-api-python and the openeo-clients couldn't discover the endpoints xD

**Pull request merged**